src/coreclr/unwinder/amd64/unwinder.cpp

```diff
@@ -257,7 +260,196 @@ BOOL OOPStackUnwinderAMD64::Unwind(CONTEXT * pContext)
     return (hr == S_OK);
 }
 
-#else // DACCESS_COMPILE
+#elif defined(FEATURE_CDAC_UNWINDER)
+
+// TODO: add asserts on cDAC build without importing more headers.
+#define UNWINDER_ASSERT(x)
+
+// Read 64 bit unsigned value from the specified addres when the unwinder is build
+// for the cDAC. This triggers a callback to the cDAC host to read the memory from
+// the target process.
+ULONG64 OOPStackUnwinderAMD64::MemoryRead64(PULONG64 addr)
+{
+    ULONG64 value;
+    readFromTarget((uint64_t)addr, &value, sizeof(value), callbackContext);
+    return value;
+}
+
+// Read 128 bit value from the specified addres when the unwinder is build
+// for the cDAC. This triggers a callback to the cDAC host to read the memory from
+// the target process.
+M128A OOPStackUnwinderAMD64::MemoryRead128(PM128A addr)
+{
+    M128A value;
+    readFromTarget((uint64_t)addr, &value, sizeof(value), callbackContext);
+    return value;
+}
+
+//---------------------------------------------------------------------------------------
+//
+// The InstructionBuffer class abstracts accessing assembler instructions in the function
+// being unwound. It behaves as a memory byte pointer, but it reads the instruction codes
+// from the target process being debugged and removes all changes that the debugger
+// may have made to the code, e.g. breakpoint instructions.
+//
+class InstructionBuffer
+{
+    UINT m_offset;
+    SIZE_T m_address;
```

Comment:
Any reason why address is of type SIZE_T and not TADDR or some other byte pointer type?